### PR TITLE
Corddis rewrite

### PR DESCRIPTION
**Pull request by tfo-dot**

### CODE REVIEW

**Comment by tfo-dot** [comment]
Main changes:

- Managers c00l ~mostly~
- Evt instead of event emitter for typed events
- Most of the intent handles were rewritten
- <Client>.get and <Guild>.get were deleted, managers instead
- Managers = Cache but better, and they're always present since they do requests like getting guild

File: src/intents/CHANNEL_DELETE.ts [hunk 1]
```diff
@@ -1,18 +1,16 @@
...
 export default async (gateway: Gateway, client: Client, data: any): Promise<any> => {
     const { guild_id } = data.d
     if (guild_id) {
-        const guild = await client.get(EntityType.GUILD, guild_id) as Guild
-        await guild.fetchChannels()
-        client.cache.guilds?.set(guild_id, guild)
+        const guild = await client.guilds.get(guild_id)
+        await guild.channels.fetchAll()
+        client.guilds.set(guild_id, guild)
         return [new ChannelStructures[data.d.type](data.d, client, guild)]
```
Comment (marek12306) [inline src/intents/CHANNEL_DELETE.ts]: this must be as object, not array

**Review by marek12306 — CHANGES_REQUESTED** [review]

File: src/structures/managers.ts [hunk 2]
```diff
@@ -62,9 +65,11 @@ export class ChannelsManager extends SubManager {
         }
     }
 
-    async fetchAll(refresh = false) {
-        if (!refresh && this.size > 0) return Array.from(this.values())
-
+    /** Fetches all channels from a guild
```
Comment (marek12306) [inline src/structures/managers.ts]: description must be in new line

File: src/structures/managers.ts [hunk 3]
```diff
@@ -49,11 +52,11 @@ export class InvitesManager extends SubManager {
 }
 
 export class ChannelsManager extends SubManager {
+    /** Fetches channel with a certain ID
```
Comment (marek12306) [inline src/structures/managers.ts]: description must be in new line

**Review by marek12306 — CHANGES_REQUESTED** [review]

**Review by marek12306 — APPROVED** [review]

**Pull request merged**